### PR TITLE
prov/efa: allocate ibv_wr for RxR provider in rxr_pkt_entry

### PR DESCRIPTION
**Pull request by sunkuamzn**
This commit is part 2 of the endpoint separation
The changes in this commit are temporary and will removed when the endpoints are fully separated

Signed-off-by: Sai Sunku <sunkusa@amazon.com>

### CODE REVIEW

File: prov/efa/src/efa_msg.c [hunk 1]
```diff
@@ -145,7 +147,11 @@ static ssize_t efa_post_recv(struct efa_ep *ep, const struct fi_msg *msg, uint64
 	ssize_t err, post_recv_err;
 	size_t i;
 
-	ewr = ofi_buf_alloc(ep->recv_wr_pool);
+	if (!msg->data) {
```
Comment (darrylabbate) [inline prov/efa/src/efa_msg.c:150]: `msg->data` can still be `NULL`/`0` even with your changes?

Comment (sunkuamzn) [reply prov/efa/src/efa_msg.c:150]: It can be NULL/0 for the dgram provider

Comment (sunkuamzn) [reply prov/efa/src/efa_msg.c:150]: I think we should prevent someone from setting `msg->data` when they directly use the DGRAM provider

Not sure show though. Adding `assert(msg->data == 0)` [here](https://github.com/ofiwg/libfabric/blob/main/prov/efa/src/efa_msg.c#L408) would affect both RDM and DGRAM.

**Comment by wzamazon** [comment]
This is not the right approach, we should looking into how to call `ibv_post_send` directly inside `rxr_pkt_entry_send`

**Review by wzamazon — CHANGES_REQUESTED** [review]
try to call `ibv_post_send/recv` diectly from RxR, bypassing efa_post_send

**Comment by sunkuamzn** [comment]
Calling `ibv_post_send`/`ibv_post_recv` directly is part 3. This PR is part 2. Should we combine part 2 with part 3? https://quip-amazon.com/x51HABgruwes/EFA-and-RXR-squash-endpoint-separation#temp:C:DHF763468a0d2384618951123c41

**Pull request closed without merging**